graphs/zero_one_bfs.py

```diff
@@ -0,0 +1,72 @@
+"""
+We all know that Breadth First Search works for unweighted graphs.
+However, it doesn't work for Weighted Graphs. Sometimes all we need
+is a 0 or 1 weight on all edges, rather than any given number.
+
+In such cases we can use this simple 0-1 BFS (Zero One Breadth First Search)
+as it is much simpler than the usual algorithm for such kind of a problem, a dijkstra.
+
+Requirement:
+Each edge of the graph must be either ZERO or ONE
+
+Refer this link: https://cp-algorithms.com/graph/01_bfs.html for more details
+"""
+
+import collections
+import math
+
+
+def zero_one_bfs(graph: dict, number_of_nodes: int, start: int, goal: int) -> None:
```

Comment:
`dict` of what?

Reply:
The way the `dict` may be formatted is given in the comments below.